### PR TITLE
Add deferrable mode to BigQueryTablePartitionExistenceSensor.

### PR DESCRIPTION
**Pull request by bkossakowska**
Add deferrable mode to BigQueryTablePartitionExistenceSensor.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/sensors/bigquery.py [hunk 1]
```diff
@@ -244,3 +247,69 @@ def execute_complete(self, context: dict[str, Any], event: dict[str, str] | None
...
+       The connection supplied to the hook must provide
+       access to the specified project.
+    :param dataset_id: The name of the dataset in which to look for the table.
+       storage bucket.
+    :param partition_id: The name of the partition to check the existence of.
+    :param table_id: The name of the table to check the existence of.
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud.
+    :param bigquery_conn_id: (Deprecated) The connection ID used to connect to Google Cloud.
+       This parameter has been deprecated. You should pass the gcp_conn_id parameter instead.
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
```
Comment (pankajastro) [inline airflow/providers/google/cloud/sensors/bigquery.py]: `delegate_to` has been deprecated so shall we avoid this adding here considering this is a new feature?

File: airflow/providers/google/cloud/sensors/bigquery.py [hunk 2]
```diff
@@ -244,3 +247,69 @@ def execute_complete(self, context: dict[str, Any], event: dict[str, str] | None
...
+       domain-wide delegation enabled.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+       credentials, or chained list of accounts required to get the access_token
+       of the last account in the list, which will be impersonated in the request.
+       If set as a string, the account must grant the originating account
+       the Service Account Token Creator IAM role.
+       If set as a sequence, the identities from the list must grant
+       Service Account Token Creator IAM role to the directly preceding identity, with first
+       account from the list granting this role to the originating account (templated).
+    :param polling_interval_seconds: The interval in seconds to wait between checks table existence.
```
Comment (pankajastro) [inline airflow/providers/google/cloud/sensors/bigquery.py]: sensors expose `poke_interval ` can we reuse that here?

File: airflow/providers/google/cloud/triggers/bigquery.py [hunk 3]
```diff
@@ -529,3 +529,67 @@ async def _table_exists(
...
+                    if records:
+                        records = [row[0] for row in records]
+                        if self.partition_id in records:
+                            yield TriggerEvent(
+                                {
+                                    "status": "success",
+                                    "message": f"Partition: {self.partition_id} in table: {self.table_id}",
+                                }
+                            )
+                            return
```
Comment (dimberman) [inline airflow/providers/google/cloud/triggers/bigquery.py]: Can you break this out into a helper function?

**Review by dimberman — APPROVED** [review]
Minor nit but overall looks good!

**Pull request merged**